RFC0012.md

```diff
@@ -0,0 +1,72 @@
+# RFC0012
+
+## Title
+Add APIs and internal support for RM-network library interactions
+
+## Abstract
+Add a network support framework and appropriate APIs so that the RM can:
+
+    * precondition an application (e.g., adding a security token to the
+      app's environment) prior to launch
+
+    * setup the local network driver to support an application (e.g., for
+      "instant on" address resolution) prior to spawning the local processes
+
+    * pass directives in the environment of client processes prior to forking
+
+    * cleanup after each child process terminates
+
+    * cleanup after all local children for a given application have
+      terminated
+
+## Labels
+[EXTENSION][SERVER-API]
+
+## Action
+
+
+## Copyright Notice
+Copyright 2016 Intel, Inc. All rights reserved.
+
+This document is subject to all provisions relating to code contributions to the PMIx community as defined in the community's [LICENSE](https://github.com/pmix/RFCs/tree/master/LICENSE) file. Code Components extracted from this document must include the License text as described in that file.
+
+## Description
+Part of the "Instant On" initiative relies on establishing a partnership between the resource manager (RM) and the networking library that allows the combination to fully setup the messaging environment prior to spawn of an application's processes. Completing this procedure enables applications to communicate without discovery and exchange of network endpoint information.
+
+There are two new APIs required to enable this support:
+
+1. Preconditioning the application for network operations. This typically involves obtaining a security token that must be used by each application process when communicating to another process in the same job. Some network libraries generate this token algorithmically, while others may need to obtain a token from a central server. Precondition values are typically passed to application processes as environmental variables that are recognized by the network library when initialized by the process. Thus, the _PMIx\_server\_setup\_application_ API takes the application's nspace (plus whatever pmix\_info\_t directives the RM chooses to provide), and returns an array of _pmix\_info\_t_ structures. The return is done via a callback function so the API will not block should the library need to obtain the token from a remote server.
+
+  The structures may consist of any combination of key-value pairs, and the RM shall:
+
+  * if the key consists of _PMIX\_SET\_ENVAR_, then the provided string shall be included in the environment of each spawned application process in this nspace
+
+  * if the key consists of _PMIX\_UNSET\_ENVAR_, then the environment of the application shall be searched, and the provided envar removed if present.
+
+  * all other key-values shall be included in the job-level info proved at process start.
```

Comment:
If I understand you correctly, you are proposing that PMIx call this function "under the covers" in register_namespace instead of asking the host server to call it directly - yes? It's a good point - I hadn't thought about it that way, but I honestly can't think of any reason not to do it internally. I can modify the RFC to that effect.